Battle/Field/field_unit.gd

```diff
@@ -76,7 +75,8 @@ func remove_target():
 func _on_input_event(viewport: Viewport, event: InputEvent, shape_idx):
 	# Listen for clicks or taps
 	# to do - allow number pad for unit target (1 targets unit1, 2 targets unit2, etc)
-	if ((event is InputEventMouseButton and event.is_pressed()) or event is InputEventScreenTouch) and !is_friendly:
+	var battleNode = get_parent().get_parent()
+	if ((event is InputEventMouseButton and event.is_pressed()) or event is InputEventScreenTouch) and !is_friendly and battleNode.zone.Stage[battleNode.current_stage-1].monsters[place_ID-1]:
```

Comment:
This line can be converted to a signal.  It looks like it is checking if a unit exists in the array. We definitely need that check, but it would be better to have `battle.gd` call a method/set a property on `field_unit` when a unit dies.

You could add another property to `filed_unit` such as `is_dead` and then have `battle.gd` set that when the state changes. This line would then check its own property, rather than having to call up the tree to get it.

Reply:
Taking care in the next Commit.

Commenting the code ⏳